<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>830</width>
    <height>672</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="canvas" name="field" native="true">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>30</y>
      <width>801</width>
      <height>451</height>
     </rect>
    </property>
   </widget>
   <widget class="QGroupBox" name="groupBox">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>490</y>
      <width>209</width>
      <height>135</height>
     </rect>
    </property>
    <property name="title">
     <string>Цвета</string>
    </property>
    <layout class="QGridLayout" name="gridLayout">
     <item row="1" column="1">
      <widget class="color_view" name="vis_part_col" native="true"/>
     </item>
     <item row="1" column="0">
      <widget class="QPushButton" name="vis_part_col_btn">
       <property name="text">
        <string>Видимая часть</string>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="color_view" name="seg_col" native="true"/>
     </item>
     <item row="2" column="1">
      <widget class="color_view" name="cut_col" native="true"/>
     </item>
     <item row="0" column="0">
      <widget class="QPushButton" name="seg_col_btn">
       <property name="text">
        <string>Исходный отрезок</string>
       </property>
      </widget>
     </item>
     <item row="2" column="0">
      <widget class="QPushButton" name="cut_col_btn">
       <property name="text">
        <string>Границы отсекателя</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QPushButton" name="pushButton">
    <property name="geometry">
     <rect>
      <x>240</x>
      <y>500</y>
      <width>211</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Отсечь</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_2">
    <property name="geometry">
     <rect>
      <x>240</x>
      <y>530</y>
      <width>211</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Очистить</string>
    </property>
   </widget>
   <widget class="QGroupBox" name="groupBox_2">
    <property name="geometry">
     <rect>
      <x>460</x>
      <y>490</y>
      <width>361</width>
      <height>135</height>
     </rect>
    </property>
    <property name="title">
     <string>Информация</string>
    </property>
    <layout class="QGridLayout" name="gridLayout_2">
     <item row="0" column="0">
      <widget class="QLabel" name="label">
       <property name="text">
        <string>Для создания горизонтальной линии зажать Shift</string>
       </property>
      </widget>
     </item>
     <item row="3" column="0">
      <widget class="QLabel" name="label_3">
       <property name="text">
        <string>Используемый метод отсечения - алгоритм Сазерленда-Ходжмена</string>
       </property>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="QLabel" name="label_2">
       <property name="text">
        <string>Для создания вертикальной линии зажать Ctrl</string>
       </property>
      </widget>
     </item>
     <item row="2" column="0">
      <widget class="QLabel" name="label_4">
       <property name="text">
        <string>Для создания линии, параллельной ребру отсекателя, зажать Alt</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QPushButton" name="end_cutter">
    <property name="geometry">
     <rect>
      <x>240</x>
      <y>560</y>
      <width>211</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Замкнуть отсекатель</string>
    </property>
   </widget>
   <widget class="QPushButton" name="end_enter">
    <property name="geometry">
     <rect>
      <x>240</x>
      <y>590</y>
      <width>211</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Замкнуть отсекаемый многоугольник</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>830</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <customwidgets>
  <customwidget>
   <class>canvas</class>
   <extends>QWidget</extends>
   <header>canvas.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>color_view</class>
   <extends>QWidget</extends>
   <header>color_view.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
